run.sh

```diff
@@ -134,11 +134,17 @@ fi
 
 ## DNS Timers
 if [ ! -z "${NETWORK_ADDRESS_CACHE_TTL}" ]; then
-    sed -i -e "s/#networkaddress.cache.ttl=.*/networkaddress.cache.ttl=${NETWORK_ADDRESS_CACHE_TTL}/" /opt/jdk-*/conf/security/java.security
+    sed -i -e "s/#networkaddress.cache.ttl=.*/networkaddress.cache.ttl=${NETWORK_ADDRESS_CACHE_TTL}/" /usr/share/elasticsearch/jdk/conf/security/java.security
 fi
 
 if [ ! -z "${NETWORK_ADDRESS_CACHE_NEGATIVE_TTL}" ]; then
-    sed -i -e ""s/networkaddress.cache.negative.ttl=.*/networkaddress.cache.negative.ttl=${NETWORK_ADDRESS_CACHE_NEGATIVE_TTL}/"" /opt/jdk-*/conf/security/java.security
+    sed -i -e ""s/networkaddress.cache.negative.ttl=.*/networkaddress.cache.negative.ttl=${NETWORK_ADDRESS_CACHE_NEGATIVE_TTL}/"" /usr/share/elasticsearch/jdk/conf/security/java.security
+fi
+
+# Is there a cluster already running or are we bootstrapping one?
+CLUSTER_RESPONSE=$(curl -I -X GET --connect-timeout 10 "${CLUSTER_URL}/_cluster/health" 2>/dev/null | head -n 1 | cut -d' ' -f 2)
+if [[ ! "$CLUSTER_RESPONSE" == "200" ]]; then
+  ES_EXTRA_ARGS+=" -Ecluster.initial_master_nodes=${MASTER_NODES}"
```

Comment:
where does "MASTER_NODES" come from ? 

Also i don't think i understand how this works 

* The first Master node coming up after a nodepool shutdown, the Response won't be 200 cause there will be no Elasticserch nodes running at that time,  health won't return and so it will set the extra args. In that case though it should not to my understanding based on [this](https://www.elastic.co/guide/en/elasticsearch/reference/current/discovery-settings.html#initial_master_nodes) since only a brand new scratch cluster should set that
* On the second node this will fail as well since while only the first master is running it will not be able to return the health endpoint since it has no quorum

What is the upstream image doing about this setting ?

Reply:
As discussed, I've added this to the master node args regardless of whether the cluster is new or not - it makes no difference.